fraction/fraction.go

```diff
@@ -0,0 +1,204 @@
+package fraction
+
+import (
+	wire "github.com/tendermint/go-wire"
+)
+
+// XXX test fractions!
+
+// Fraction -  basic fraction functionality
+// TODO better name that Fraction?
+type Fraction interface {
+	GetNumerator() int64
+	GetDenominator() int64
+	RectifySign() Fraction
+	Inv() Fraction
+	Simplify() Fraction
+	Negative() bool
+	Positive() bool
+	GT(Fraction) bool
+	LT(Fraction) bool
+	Equal(Fraction) bool
+	Mul(Fraction) Fraction
+	Div(Fraction) Fraction
+	Add(Fraction) Fraction
+	Sub(Fraction) Fraction
+	Evaluate() int64
+}
+
+// fraction - basic fraction
+type fraction struct {
+	Numerator, Denominator int64
+}
+
+var _ Fraction = fraction{} // enforce at compile time
+var _ = wire.RegisterInterface(struct{ Fraction }{}, wire.ConcreteType{fraction{}, 0x01})
+
+// New - create a new fraction object
+func New(Numerator int64, Denominator ...int64) Fraction {
+	switch len(Denominator) {
+	case 0:
+		return fraction{Numerator, 1}
+	case 1:
+		return fraction{Numerator, Denominator[0]}
+	default:
+		panic("improper use of NewFraction, can only have one denominator")
+	}
+}
+
+// GetNumerator - return the Numerator
+func (f fraction) GetNumerator() int64 {
+	return f.Numerator
+}
+
+// GetDenominator - return the Denominator
+func (f fraction) GetDenominator() int64 {
+	return f.Denominator
+}
+
+// RectifySign - make any negative sign exist only in Numerator
+func (f fraction) RectifySign() Fraction {
+	if f.Denominator < 0 {
+		return fraction{-1 * f.Numerator, -1 * f.Denominator}
+	}
+	return f
+}
+
+// Inv - Inverse
+func (f fraction) Inv() Fraction {
+	return fraction{f.Denominator, f.Numerator}
+}
+
+// Simplify - find the greatest common Denominator, divide
+func (f fraction) Simplify() Fraction {
+	gcd := f.Numerator
+	for d := f.Denominator; d != 0; {
+		gcd, d = d, gcd%d
+	}
+	f.Numerator /= gcd
+	f.Denominator /= gcd
+	return f.RectifySign()
+}
+
+// Negative - is the fractior negative
+func (f fraction) Negative() bool {
+	switch {
+	case f.Numerator > 0:
+		if f.Denominator > 0 {
+			return false
+		}
+		return true
```

Comment:
so if Denominator == 0, we return true? not saying this is incorrect. just checking

Reply:
why not just `!f.Positive()`

Reply:
because of the zero case, if zero then the result is neither neg nor positive........ Antiquated though